### PR TITLE
Fix the elapseRate in Ingestion Test and Polish Existing Code 

### PR DESCRIPTION
**Pull request by liuruiyiyang**
Change List:
1. Use workload value buffer for more accurate ```elapseRate``` calculation in ingestion test.
2. Remove the ```accRate``` in the ```Result Matrix```
3. Output test progress log at fixed interval to prevent too much log which may affect the test result(elapse time).
4. Unify the timing to the DBWrapper level instead of inside each DB's implementation
5. Add a Diagnosis class for time cost evaluation.
6. Update README
7. Fix the format of result issue #34

### CODE REVIEW

File: src/main/java/cn/edu/tsinghua/iotdb/benchmark/workload/SyntheticWorkload.java [hunk 1]
```diff
@@ -48,6 +49,24 @@ public SyntheticWorkload(int clientId) {
     }
   }
 
+  private static String[][] initWorkloadValues () {
+    String[][] workloadValues = null;
+    if(!config.OPERATION_PROPORTION.split(":")[0].equals("0")) {
+      workloadValues = new String[config.SENSOR_NUMBER][config.BATCH_EXECUTE_COUNT];
```
Comment (qiaojialin) [inline src/main/java/cn/edu/tsinghua/iotdb/benchmark/workload/SyntheticWorkload.java]: should here use BATCH_SIZE ? What is BATCH_EXECUTE_COUNT mean?

File: src/main/java/cn/edu/tsinghua/iotdb/benchmark/workload/SyntheticWorkload.java [hunk 2]
```diff
@@ -102,21 +121,17 @@ private Batch getDistOutOfOrderBatch(DeviceSchema deviceSchema) {
...
-    long currentTimestamp;
-    currentTimestamp = getCurrentTimestamp(stepOffset);
-    for (String sensor : deviceSchema.getSensors()) {
-      FunctionParam param = config.SENSOR_FUNCTION.get(sensor);
-      String value = String.format(DECIMAL_FORMAT,
-          Function.getValueByFuntionidAndParam(param, currentTimestamp).floatValue());
-      values.add(value);
+    long currentTimestamp = getCurrentTimestamp(stepOffset);
+    for(int i = 0;i < config.SENSOR_NUMBER;i++) {
+      values.add(workloadValues[i][(int)(stepOffset % config.BATCH_EXECUTE_COUNT)]);
```
Comment (qiaojialin) [inline src/main/java/cn/edu/tsinghua/iotdb/benchmark/workload/SyntheticWorkload.java]: also, why use BATCH_EXECUTE_COUNT?

Comment (liuruiyiyang) [reply src/main/java/cn/edu/tsinghua/iotdb/benchmark/workload/SyntheticWorkload.java]: fixed, use WORKLOAD_BUFFER_SIZE

File: src/main/java/cn/edu/tsinghua/iotdb/benchmark/workload/SyntheticWorkload.java (hunk 1 above)
Comment (liuruiyiyang) [reply src/main/java/cn/edu/tsinghua/iotdb/benchmark/workload/SyntheticWorkload.java]: fixed, use WORKLOAD_BUFFER_SIZE

File: src/main/java/cn/edu/tsinghua/iotdb/benchmark/client/BaseClient.java [hunk 3]
```diff
@@ -33,28 +39,33 @@ public BaseClient(int id, CountDownLatch countDownLatch, CyclicBarrier barrier,
     singletonWorkload = SingletonWorkload.getInstance();
     operationController = new OperationController(id);
     insertLoopIndex = 0;
-    initLogger();
   }
 
   void doTest() {
-    for (long loopIndex = 0; loopIndex < config.LOOP; loopIndex++) {
+    String currentThread = Thread.currentThread().getName();
+    service.scheduleAtFixedRate(() -> {
```
[the remark this answers is not in the document]
Comment (liuruiyiyang) [reply src/main/java/cn/edu/tsinghua/iotdb/benchmark/client/BaseClient.java]: added

File: README.md [hunk 4]
```diff
@@ -256,59 +256,58 @@ Now after launching the test, you will see testing information rolling like foll
...
+AGG_RANGE_VALUE     2327                2327                0                   0                   20.62               
+GROUP_BY            2283                29679               0                   0                   262.95              
+LATEST_POINT        2206                2206                0                   0                   19.54               
+---------------------------------------------------------------------------------------------------------------------------------
+--------------------------------------------------------------------------Latency (ms) Matrix--------------------------------------------------------------------------
+Operation           AVG         MID_AVG     MIN         P10         P25         MEDIAN      P75         P90         P95         P99         MAX         MAX_SUM     
+INGESTION           0.00        0.00        0.00        0.00        0.00        0.00        0.00        0.00        0.00        0.00        0.00        0.00        
+PRECISE_POINT       67.98       67.36       6.07        37.51       49.49       56.64       96.85       100.43      112.30      149.98      197.27      8466.59
```
[the remark this answers is not in the document]
Comment (liuruiyiyang) [reply README.md]: Yes

File: README.md [hunk 5]
```diff
@@ -155,32 +155,32 @@ OVERFLOW_MODE: 0
...
+VALUE_RANGE         0                   0                   0                   0                   0.00                
+AGG_RANGE           0                   0                   0                   0                   0.00                
+AGG_VALUE           0                   0                   0                   0                   0.00                
+AGG_RANGE_VALUE     0                   0                   0                   0                   0.00                
+GROUP_BY            0                   0                   0                   0                   0.00                
+LATEST_POINT        0                   0                   0                   0                   0.00                
+---------------------------------------------------------------------------------------------------------------------------------
+--------------------------------------------------------------------------Latency (ms) Matrix--------------------------------------------------------------------------
+Operation           AVG         MID_AVG     MIN         P10         P25         MEDIAN      P75         P90         P95         P99         MAX         MAX_SUM
```
[the remark this answers is not in the document]
Comment (liuruiyiyang) [reply README.md]: changed to SLOWEST_THREAD

File: README.md [hunk 6]
```diff
@@ -192,12 +192,12 @@ The output contains overall information of the test including:
 + failOperation: the request/SQL number failed to execute for different operations
 + failPoint: the data point number failed to ingest (for query operations currently this field is always zero)
 + elapseRate: equals to ```okPoint / Test elapse time```
```
[the remark this answers is not in the document]
Comment (liuruiyiyang) [reply README.md]: changed to throughput

**Pull request merged**